driver-core/src/main/com/mongodb/internal/connection/DescriptionHelper.java

```diff
@@ -144,6 +147,11 @@ private static TopologyVersion getTopologyVersion(final BsonDocument isMasterRes
                 ? new TopologyVersion(isMasterResult.getDocument("topologyVersion")) : null;
     }
 
+    private static ObjectId getServiceId(final BsonDocument isMasterResult) {
+        return isMasterResult.containsKey("serviceId") && isMasterResult.get("serviceId").isObjectId()
+                ? isMasterResult.getObjectId("serviceId").getValue() : null;
```

Comment:
[optional]

This code does 3 map lookups instead of 1. This does not affect correctness, though.